Enable `dupl ` in the linter and fix issues

## Feature Request

This issue is for we track the need to enable  `--enable= dupl `. See: https://github.com/operator-framework/operator-sdk/blob/master/hack/go-linter.sh#L13

We should do a PR with the check enabled and its fixes. Following the current erro:

```
 $ make test-linter
./hack/go-linter.sh
Checking if golangci-lint is installed
Running golangci-lint
internal/scorecard/plugins/olm_tests.go:383: 383-419 lines are duplicate of `internal/scorecard/plugins/olm_tests.go:422-458` (dupl)
func (t *StatusDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
	res := &schelpers.TestResult{Test: t}
	err := t.Client.Get(ctx, types.NamespacedName{Namespace: t.CR.GetNamespace(), Name: t.CR.GetName()}, t.CR)
	if err != nil {
		res.Errors = append(res.Errors, err)
		res.State = scapiv1alpha1.ErrorState
		return res
	}
	if t.CR.Object["status"] == nil {
		return res
	}
	statusBlock := t.CR.Object["status"].(map[string]interface{})
	res.MaximumPoints = len(statusBlock)
	var crd *olmapiv1alpha1.CRDDescription
	for _, owned := range t.CSV.Spec.CustomResourceDefinitions.Owned {
		if owned.Kind == t.CR.GetKind() {
			crd = &owned
			break
		}
	}
	if crd == nil {
		return res
	}
	for key := range statusBlock {
		for _, statDesc := range crd.StatusDescriptors {
			if statDesc.Path == key {
				res.EarnedPoints++
				delete(statusBlock, key)
				break
			}
		}
	}
	for key := range statusBlock {
		res.Suggestions = append(res.Suggestions, "Add a status descriptor for "+key)
	}
	return res
}
internal/scorecard/plugins/olm_tests.go:422: 422-458 lines are duplicate of `internal/scorecard/plugins/olm_tests.go:383-419` (dupl)
func (t *SpecDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
	res := &schelpers.TestResult{Test: t}
	err := t.Client.Get(ctx, types.NamespacedName{Namespace: t.CR.GetNamespace(), Name: t.CR.GetName()}, t.CR)
	if err != nil {
		res.Errors = append(res.Errors, err)
		res.State = scapiv1alpha1.ErrorState
		return res
	}
	if t.CR.Object["spec"] == nil {
		return res
	}
	specBlock := t.CR.Object["spec"].(map[string]interface{})
	res.MaximumPoints = len(specBlock)
	var crd *olmapiv1alpha1.CRDDescription
	for _, owned := range t.CSV.Spec.CustomResourceDefinitions.Owned {
		if owned.Kind == t.CR.GetKind() {
			crd = &owned
			break
		}
	}
	if crd == nil {
		return res
	}
	for key := range specBlock {
		for _, statDesc := range crd.SpecDescriptors {
			if statDesc.Path == key {
				res.EarnedPoints++
				delete(specBlock, key)
				break
			}
		}
	}
	for key := range specBlock {
		res.Suggestions = append(res.Suggestions, "Add a spec descriptor for "+key)
	}
	return res
}
make: *** [test-linter] Error 1
```

--- a/hack/go-linter.sh
+++ b/hack/go-linter.sh
@@ -21,11 +21,7 @@ golangci-lint run --disable-all \
 # --enable=gocyclo
 # --enable=lll
 # --enable=gosec
-<<<<<<< HEAD
-=======
 # --enable=misspell \
-# --enable=errcheck \
->>>>>>> 64f6f0f8... feat(linter): remove linter comment
 # --enable=interfacer \
 # --enable=misspell \
 # --enable=varcheck \

--- a/hack/go-linter.sh
+++ b/hack/go-linter.sh
@@ -14,13 +14,13 @@ golangci-lint run --disable-all \
     --enable=goconst \
     --enable=goimports \
     --enable=errcheck \
+    --enable=dupl \
 
 ##todo(camilamacedo86): The following checks requires fixes in the code
 # --enable=golint
 # --enable=gocyclo
 # --enable=lll
 # --enable=gosec
-# --enable=dupl \
 # --enable=interfacer \
 # --enable=misspell \
 # --enable=varcheck \

--- a/hack/go-linter.sh
+++ b/hack/go-linter.sh
@@ -21,6 +21,11 @@ golangci-lint run --disable-all \
 # --enable=gocyclo
 # --enable=lll
 # --enable=gosec
+<<<<<<< HEAD
+=======
+# --enable=misspell \
+# --enable=errcheck \
+>>>>>>> 64f6f0f8... feat(linter): remove linter comment
 # --enable=interfacer \
 # --enable=misspell \
 # --enable=varcheck \

--- a/internal/scaffold/role.go
+++ b/internal/scaffold/role.go
@@ -89,22 +89,8 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 			pr.Verbs = []string{"*"}
 			role.Rules = append(role.Rules, *pr)
 		}
-		// update role.yaml
-		d, err := json.Marshal(&role)
-		if err != nil {
-			return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
-		}
-		m := &map[string]interface{}{}
-		if err := yaml.Unmarshal(d, m); err != nil {
-			return fmt.Errorf("failed to unmarshal role(%+v): %v", role, err)
-		}
-		data, err := yaml.Marshal(m)
-		if err != nil {
-			return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
-		}
-		if err := ioutil.WriteFile(roleFilePath, data, fileutil.DefaultFileMode); err != nil {
-			return fmt.Errorf("failed to update %v: %v", roleFilePath, err)
-		}
+
+		return updateYamlForRole(&role, roleFilePath)
 	case *rbacv1.ClusterRole:
 		pr := &rbacv1.PolicyRule{}
 		apiGroupFound := false
@@ -133,27 +119,30 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 			pr.Verbs = []string{"*"}
 			role.Rules = append(role.Rules, *pr)
 		}
-		// update role.yaml
-		d, err := json.Marshal(&role)
-		if err != nil {
-			return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
-		}
-		m := &map[string]interface{}{}
-		if err = yaml.Unmarshal(d, m); err != nil {
-			return fmt.Errorf("failed to unmarshal role(%+v): %v", role, err)
-		}
 
-		data, err := yaml.Marshal(m)
-		if err != nil {
-			return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
-		}
-		if err := ioutil.WriteFile(roleFilePath, data, fileutil.DefaultFileMode); err != nil {
-			return fmt.Errorf("failed to update %v: %v", roleFilePath, err)
-		}
+		return updateYamlForRole(&role, roleFilePath)
 	default:
 		return errors.New("failed to parse role.yaml as a role")
 	}
-	// not reachable
+}
+
+func updateYamlForRole(role interface{}, roleFilePath string) error {
+	d, err := json.Marshal(&role)
+	if err != nil {
+		return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
+	}
+	m := &map[string]interface{}{}
+	if err := yaml.Unmarshal(d, m); err != nil {
+		return fmt.Errorf("failed to unmarshal role(%+v): %v", role, err)
+	}
+	data, err := yaml.Marshal(m)
+	if err != nil {
+		return fmt.Errorf("failed to marshal role(%+v): %v", role, err)
+	}
+	if err := ioutil.WriteFile(roleFilePath, data, fileutil.DefaultFileMode); err != nil {
+		return fmt.Errorf("failed to update %v: %v", roleFilePath, err)
+	}
+
 	return nil
 }
 

--- a/internal/scaffold/role.go
+++ b/internal/scaffold/role.go
@@ -90,7 +90,7 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 			role.Rules = append(role.Rules, *pr)
 		}
 
-		return updateYamlForRole(&role, roleFilePath)
+		return updateRoleFile(&role, roleFilePath)
 	case *rbacv1.ClusterRole:
 		pr := &rbacv1.PolicyRule{}
 		apiGroupFound := false
@@ -120,13 +120,13 @@ func UpdateRoleForResource(r *Resource, absProjectPath string) error {
 			role.Rules = append(role.Rules, *pr)
 		}
 
-		return updateYamlForRole(&role, roleFilePath)
+		return updateRoleFile(&role, roleFilePath)
 	default:
 		return errors.New("failed to parse role.yaml as a role")
 	}
 }
 
-func updateYamlForRole(role interface{}, roleFilePath string) error {
+func updateRoleFile(role interface{}, roleFilePath string) error {
 	d, err := json.Marshal(&role)
 	if err != nil {
 		return fmt.Errorf("failed to marshal role(%+v): %v", role, err)

--- a/internal/scorecard/plugins/olm_tests.go
+++ b/internal/scorecard/plugins/olm_tests.go
@@ -33,6 +33,11 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const (
+	statusDescriptor string = "status"
+	specDescriptor   string = "spec"
+)
+
 // OLMTestConfig contains all variables required by the OLMTest schelpers.TestSuite
 type OLMTestConfig struct {
 	Client   client.Client
@@ -394,34 +399,8 @@ func (t *StatusDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
 		res.State = scapiv1alpha1.ErrorState
 		return res
 	}
-	if t.CR.Object["status"] == nil {
-		return res
-	}
-	statusBlock := t.CR.Object["status"].(map[string]interface{})
-	res.MaximumPoints = len(statusBlock)
-	var crd *olmapiv1alpha1.CRDDescription
-	for _, owned := range t.CSV.Spec.CustomResourceDefinitions.Owned {
-		if owned.Kind == t.CR.GetKind() {
-			crd = &owned
-			break
-		}
-	}
-	if crd == nil {
-		return res
-	}
-	for key := range statusBlock {
-		for _, statDesc := range crd.StatusDescriptors {
-			if statDesc.Path == key {
-				res.EarnedPoints++
-				delete(statusBlock, key)
-				break
-			}
-		}
-	}
-	for key := range statusBlock {
-		res.Suggestions = append(res.Suggestions, "Add a status descriptor for "+key)
-	}
-	return res
+
+	return checkCRConfiguration(t.CR, t.CSV, statusDescriptor, res)
 }
 
 // Run - implements Test interface
@@ -433,32 +412,53 @@ func (t *SpecDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
 		res.State = scapiv1alpha1.ErrorState
 		return res
 	}
-	if t.CR.Object["spec"] == nil {
+
+	return checkCRConfiguration(t.CR, t.CSV, specDescriptor, res)
+}
+
+func checkCRConfiguration(cr *unstructured.Unstructured, csv *olmapiv1alpha1.ClusterServiceVersion, descriptor string, res *schelpers.TestResult) *schelpers.TestResult {
+	if cr.Object[descriptor] == nil {
 		return res
 	}
-	specBlock := t.CR.Object["spec"].(map[string]interface{})
-	res.MaximumPoints = len(specBlock)
+	block := cr.Object[descriptor].(map[string]interface{})
+	res.MaximumPoints = len(block)
 	var crd *olmapiv1alpha1.CRDDescription
-	for _, owned := range t.CSV.Spec.CustomResourceDefinitions.Owned {
-		if owned.Kind == t.CR.GetKind() {
+	for _, owned := range csv.Spec.CustomResourceDefinitions.Owned {
+		if owned.Kind == cr.GetKind() {
 			crd = &owned
 			break
 		}
 	}
+
 	if crd == nil {
 		return res
 	}
-	for key := range specBlock {
-		for _, statDesc := range crd.SpecDescriptors {
-			if statDesc.Path == key {
-				res.EarnedPoints++
-				delete(specBlock, key)
-				break
+
+	if descriptor == statusDescriptor {
+		for key := range block {
+			for _, statDesc := range crd.StatusDescriptors {
+				if statDesc.Path == key {
+					res.EarnedPoints++
+					delete(block, key)
+					break
+				}
 			}
 		}
 	}
-	for key := range specBlock {
-		res.Suggestions = append(res.Suggestions, "Add a spec descriptor for "+key)
+	if descriptor == specDescriptor {
+		for key := range block {
+			for _, statDesc := range crd.SpecDescriptors {
+				if statDesc.Path == key {
+					res.EarnedPoints++
+					delete(block, key)
+					break
+				}
+			}
+		}
+	}
+
+	for key := range block {
+		res.Suggestions = append(res.Suggestions, fmt.Sprintf("Add a %s descriptor for %s", descriptor, key))
 	}
 	return res
 }

--- a/internal/scorecard/plugins/olm_tests.go
+++ b/internal/scorecard/plugins/olm_tests.go
@@ -447,8 +447,8 @@ func checkCRConfiguration(cr *unstructured.Unstructured, csv *olmapiv1alpha1.Clu
 	}
 	if descriptor == specDescriptor {
 		for key := range block {
-			for _, statDesc := range crd.SpecDescriptors {
-				if statDesc.Path == key {
+			for _, specDesc := range crd.SpecDescriptors {
+				if specDesc.Path == key {
 					res.EarnedPoints++
 					delete(block, key)
 					break

--- a/internal/scorecard/plugins/olm_tests.go
+++ b/internal/scorecard/plugins/olm_tests.go
@@ -400,7 +400,7 @@ func (t *StatusDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
 		return res
 	}
 
-	return checkCRConfiguration(t.CR, t.CSV, statusDescriptor, res)
+	return checkOwnedCSVDescriptors(t.CR, t.CSV, statusDescriptor, res)
 }
 
 // Run - implements Test interface
@@ -413,10 +413,10 @@ func (t *SpecDescriptorsTest) Run(ctx context.Context) *schelpers.TestResult {
 		return res
 	}
 
-	return checkCRConfiguration(t.CR, t.CSV, specDescriptor, res)
+	return checkOwnedCSVDescriptors(t.CR, t.CSV, specDescriptor, res)
 }
 
-func checkCRConfiguration(cr *unstructured.Unstructured, csv *olmapiv1alpha1.ClusterServiceVersion, descriptor string, res *schelpers.TestResult) *schelpers.TestResult {
+func checkOwnedCSVDescriptors(cr *unstructured.Unstructured, csv *olmapiv1alpha1.ClusterServiceVersion, descriptor string, res *schelpers.TestResult) *schelpers.TestResult {
 	if cr.Object[descriptor] == nil {
 		return res
 	}

